dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/ResultTable.scala

```diff
@@ -55,44 +55,40 @@ class ResultTable(
   *   this is the main result - a known path
   * @param table
   *   the result table - kept to allow for detailed inspection of intermediate paths
+  * @param resultType
+  *   see ResultType below
   * @param callSiteStack
   *   the call site stack containing the call sites that were expanded to kick off the task. We require this to match
   *   call sites to exclude non-realizable paths through other callers
-  * @param partial
-  *   indicate whether this result stands on its own or requires further analysis, e.g., by expanding output arguments
-  *   backwards into method output parameters.
   */
 case class ReachableByResult(
   path: Vector[PathElement],
   table: ResultTable,
-  callSiteStack: mutable.Stack[Call],
-  callDepth: Int = 0,
-  partial: Boolean = false
+  callSiteStack: List[Call], // immutable.Stack is Deprecated in scala 2.13. Use a List instead
+  resultType: ResultType = ReachSource,
+  callDepth: Int = 0
 ) {
   def startingPoint: CfgNode = path.head.node
-
```

Comment:
Don't need this any more. Because we put the ```resultType``` in the ```ReachableByResult```, if the ```resultType``` is ```ReachCall```, we know for sure the head element is a ```Call```.